include/net/aws_iot.h

```diff
@@ -187,6 +187,10 @@ struct aws_iot_config {
 	char *client_id;
 	/** Length of client_id string. */
 	size_t client_id_len;
+#if defined(CONFIG_AWS_IOT_BROKER_HOST_NAME_APP)
+	/** AWS IoT endpoint host name for broker connection */
+	char *host_name;
```

Comment:
Either document that the passed in `host_name` is expected to be null terminated or add a length parameter that forces the user to set the length when calling the function. I think I would prefer the latter to align with `client_id`.